### ISSUE TITLE
VSCode terminal stops showing input after running CLI

### ISSUE THREAD

**Issue by cpboyd** [gatsbyjs/gatsby#16520]
## Description

I can't see anything that I type after running `gatsby` from the VSCode terminal.

### Steps to reproduce

1. Run `gatsby develop` and hit Ctrl+C.
2. Terminal stops showing text input.

### Expected result

Text input should still display after running `gatsby develop`.

### Actual result

Text input is not shown.

### Environment

```
  System:
    OS: macOS 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.8.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.10.3 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 76.0.3809.100
    Safari: 12.1.2
  npmPackages:
    gatsby: ^2.13.57 => 2.13.57
    gatsby-image: ^2.2.8 => 2.2.8
    gatsby-plugin-catch-links: ^2.1.2 => 2.1.2
    gatsby-plugin-google-analytics: ^2.1.7 => 2.1.7
    gatsby-plugin-layout: ^1.1.3 => 1.1.3
    gatsby-plugin-manifest: ^2.2.5 => 2.2.5
    gatsby-plugin-offline: ^2.2.5 => 2.2.5
    gatsby-plugin-react-helmet: ^3.1.3 => 3.1.3
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0
    gatsby-plugin-sass: ^2.1.6 => 2.1.6
    gatsby-plugin-sharp: ^2.2.11 => 2.2.11
    gatsby-plugin-sitemap: ^2.2.6 => 2.2.6
    gatsby-plugin-typescript: ^2.1.2 => 2.1.2
    gatsby-remark-embed-video: ^1.7.1 => 1.7.1
    gatsby-remark-external-links: ^0.0.4 => 0.0.4
    gatsby-remark-images: ^3.1.8 => 3.1.8
    gatsby-remark-responsive-iframe: ^2.2.4 => 2.2.4
    gatsby-source-filesystem: ^2.1.9 => 2.1.9
    gatsby-transformer-json: ^2.2.2 => 2.2.2
    gatsby-transformer-remark: ^2.6.12 => 2.6.12
    gatsby-transformer-sharp: ^2.2.6 => 2.2.6
  npmGlobalPackages:
    gatsby-cli: 2.7.29
```

**Comment by freiksenet** [comment]
CC @wardpeet

**Comment by m-allanson** [comment]
@cpboyd I was unable to replicate this with VSCode 1.36.1 or 1.37.0. I can keep typing after hitting Ctrl+C in the VS Code terminal. 

Do you have access to a second computer that you're able to replicate this on? Does this only happen in the VS Code terminal or can you recreate it using Terminal.app?

<details>
<summary>Click here to view terminal output</summary>

```
ᛝ ~/d/temp/
gatsby new vscode-terminal-test
info Creating new site from git:

Cloning into 'vscode-terminal-test'...
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 2001 (delta 0), reused 1 (delta 0), pack-reused 1996
Receiving objects: 100% (2001/2001), 8.98 MiB | 5.26 MiB/s, done.
success Created starter directory layout
info Installing packages...

yarn install v1.17.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
info Initialising git in vscode-terminal-test

info Create initial git commit in vscode-terminal-test

ᛝ ~/d/temp/
cd vscode-terminal-test/
ᛝ ~/../temp/vscode-terminal-test/ ᚠ master
gatsby develop
success open and validate gatsby-configs - 0.086 s
success load plugins - 0.358 s
success onPreInit - 0.043 s
success initialize cache - 0.040 s
success copy gatsby files - 0.069 s
success onPreBootstrap - 0.041 s
success source and transform nodes - 0.172 s
success building schema - 0.217 s
success createPages - 0.033 s
success createPagesStatefully - 0.084 s
success onPreExtractQueries - 0.039 s
success update schema - 0.068 s
success extract queries from components - 0.285 s
success write out requires - 0.062 s
success write out redirect data - 0.040 s
success Build manifest and related icons - 0.159 s
success onPostBootstrap - 0.206 s
⠀
info bootstrap finished - 4.139 s
⠀
success run static queries - 0.230 s — 3/3 16.94 queries/second
success run page queries - 0.102 s — 5/5 99.46 queries/second
success Generating image thumbnails — 6/6 - 0.410 s
 DONE  Compiled successfully in 3047ms                                    10:10:29 AM
⠀
You can now view gatsby-starter-default in the browser.
⠀
  http://localhost:8000/
⠀
View GraphiQL, an in-browser IDE, to explore your site's data and schema
⠀
  http://localhost:8000/___graphql
⠀
Note that the development build is not optimized.
To create a production build, use npm run build
⠀
ℹ ｢wdm｣:
ℹ ｢wdm｣: Compiled successfully.


—————————————————————————————————————————————————————————————————————————————————————
5 pages                                                        gatsby-starter-default
ᛝ ~/../temp/vscode-terminal-test/ ᚠ master
gatsby info

  System:
    OS: macOS 10.14.6
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Languages:
    Python: 3.5.3 - /Users/mike/.pyenv/shims/python
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 68.0.1
    Safari: 12.1.2
  npmPackages:
    gatsby: ^2.13.58 => 2.13.58
    gatsby-image: ^2.2.8 => 2.2.8
    gatsby-plugin-manifest: ^2.2.5 => 2.2.5
    gatsby-plugin-offline: ^2.2.5 => 2.2.5
    gatsby-plugin-react-helmet: ^3.1.3 => 3.1.3
    gatsby-plugin-sharp: ^2.2.11 => 2.2.11
    gatsby-source-filesystem: ^2.1.9 => 2.1.9
    gatsby-transformer-sharp: ^2.2.6 => 2.2.6
  npmGlobalPackages:
    gatsby-cli: 2.7.30
    gatsby-dev-cli: 2.4.19
```

</details>

**Comment by krishnakakade1999** [comment]
i think you all try to reinstall it and 
<b>and after that execute this commands </b>
<b>sudo get apt</b>
<b>sudo get update gatsby</b>
<b>try this if not working then let me know</b>

**Comment by freiksenet** [comment]
> i think you all try to reinstall it and
> and after that execute this commands 
> sudo get apt
> sudo get update gatsby
> try this if not working then let me know

Hi @krishnakakade1999! 

I'm not sure I understand your answer, are you sure that it applies to this issue? 

Thanks!

**Comment by krishnakakade1999** [comment]
@freiksenet try it it  is worked for my friend give it a try then we will see ok

**Comment by cpboyd** [comment]
@m-allanson I just tried a clean install on Ubuntu 16.04 with VSCode 1.37.0 and same issue.

I'll try to work on a minimal reproduction.

VSCode:
```
Version: 1.37.0
Commit: 036a6b1d3ac84e5ca96a17a44e63a87971f8fcc8
Date: 2019-08-08T01:24:14.598Z
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Linux x64 4.15.0-55-generic
```
Gatsby:
```

  System:
    OS: Linux 4.15 Ubuntu 16.04.6 LTS (Xenial Xerus)
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 4.3.48 - /bin/bash
  Binaries:
    Node: 12.8.0 - /usr/bin/node
    Yarn: 1.17.3 - /usr/bin/yarn
    npm: 6.10.2 - /usr/bin/npm
  Languages:
    Python: 2.7.12 - /usr/bin/python
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 68.0.1
  npmPackages:
    gatsby: ^2.9.4 => 2.10.4 
    gatsby-image: ^2.1.4 => 2.2.3 
    gatsby-plugin-catch-links: ^2.0.15 => 2.1.0 
    gatsby-plugin-google-analytics: ^2.0.21 => 2.1.0 
    gatsby-plugin-layout: ^1.0.15 => 1.1.0 
    gatsby-plugin-manifest: ^2.1.1 => 2.2.0 
    gatsby-plugin-offline: ^2.1.1 => 2.2.0 
    gatsby-plugin-react-helmet: ^3.0.12 => 3.1.0 
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0 
    gatsby-plugin-sass: ^2.0.11 => 2.1.0 
    gatsby-plugin-sharp: ^2.1.3 => 2.2.1 
    gatsby-plugin-sitemap: ^2.1.0 => 2.2.0 
    gatsby-plugin-typescript: ^2.0.15 => 2.1.0 
    gatsby-remark-embed-video: ^1.7.1 => 1.7.1 
    gatsby-remark-external-links: ^0.0.4 => 0.0.4 
    gatsby-remark-images: ^3.0.14 => 3.1.0 
    gatsby-remark-responsive-iframe: ^2.1.1 => 2.2.0 
    gatsby-source-filesystem: ^2.0.39 => 2.1.0 
    gatsby-transformer-json: ^2.1.11 => 2.2.0 
    gatsby-transformer-remark: ^2.3.12 => 2.5.0 
    gatsby-transformer-sharp: ^2.1.21 => 2.2.0 
  npmGlobalPackages:
    gatsby: 2.13.59
```

**Comment by cpboyd** [comment]
@m-allanson I basically removed everything from my blog and kept running into the issue...

So I created a new blog from https://github.com/gatsbyjs/gatsby-starter-blog and still had the issue.
Followed the instructions, selected Yarn.

```
  System:
    OS: Linux 4.15 Ubuntu 16.04.6 LTS (Xenial Xerus)
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 4.3.48 - /bin/bash
  Binaries:
    Node: 12.8.0 - /usr/bin/node
    Yarn: 1.17.3 - /usr/bin/yarn
    npm: 6.10.2 - /usr/bin/npm
  Languages:
    Python: 2.7.12 - /usr/bin/python
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 68.0.1
  npmPackages:
    gatsby: ^2.13.61 => 2.13.61 
    gatsby-image: ^2.2.8 => 2.2.8 
    gatsby-plugin-feed: ^2.3.6 => 2.3.6 
    gatsby-plugin-google-analytics: ^2.1.7 => 2.1.7 
    gatsby-plugin-manifest: ^2.2.5 => 2.2.5 
    gatsby-plugin-offline: ^2.2.5 => 2.2.5 
    gatsby-plugin-react-helmet: ^3.1.3 => 3.1.3 
    gatsby-plugin-sharp: ^2.2.11 => 2.2.11 
    gatsby-plugin-typography: ^2.3.2 => 2.3.2 
    gatsby-remark-copy-linked-files: ^2.1.5 => 2.1.5 
    gatsby-remark-images: ^3.1.10 => 3.1.10 
    gatsby-remark-prismjs: ^3.3.5 => 3.3.5 
    gatsby-remark-responsive-iframe: ^2.2.4 => 2.2.4 
    gatsby-remark-smartypants: ^2.1.2 => 2.1.2 
    gatsby-source-filesystem: ^2.1.9 => 2.1.9 
    gatsby-transformer-remark: ^2.6.13 => 2.6.13 
    gatsby-transformer-sharp: ^2.2.6 => 2.2.6 
  npmGlobalPackages:
    gatsby: 2.13.59
```

**Comment by infomofo** [comment]
I'm having the same issue on a MacOS iTerm/tmux setup. After I hit CTRL-C during a gatsby develop session my terminal stops echoing any of the characters I'm typing. The commands I type still work but it's very frustrating.

**Comment by rjray** [comment]
I believe this is the same issue that I reported in #16621, which has a pending fix. The CLI program doesn't reset the terminal state before exiting on a Ctrl-C.

If you type `reset` and hit return, the terminal should reset the TTY and return to normal.

**Comment by wardpeet** [comment]
When #16621 is updated I'll see if it fixes the problem. I've noticed this bug before as well, I thought it was a vscode issue as I've never seen this in PowerShell itself.

**Comment by Js-Brecht** [comment]
This should have been fixed by #16714

**Comment by cpboyd** [comment]
@Js-Brecht I'm no longer seeing the issue.
